### ISSUE TITLE
smoke: solaris, g++, flock

### ISSUE THREAD

**Issue by hvds** [Perl/perl5#17482]
Smoke reports for g++ (versions ranging from 6.4.0 to 7.4.0) on Solaris 2.11 appear to have been added around 2018-09-08 (first appearing at commit 70fe82288ca1), and have always failed since that time with:
```
Undefined			first referenced
 symbol  			    in file
flock(int, int)            pp_sys.o
ld: fatal: symbol referencing errors. No output written to miniperl
```

If we could fix that, it would improve the value of the smoke reports.

Hugo

**Comment by khwilliamson** [comment]
On 1/21/20 8:40 AM, Hugo van der Sanden wrote:
> Smoke reports for g++ (versions ranging from 6.4.0 to 7.4.0) on Solaris 
> 2.11 appear to have been added around 2018-09-08 (first appearing at 
> commit 70fe822 
> <https://github.com/Perl/perl5/commit/70fe82288ca1a795b49f88f679cc3ef99e6f1422>), 
> and have always failed since that time with:
> 
> |Undefined first referenced symbol in file flock(int, int) pp_sys.o ld: 
> fatal: symbol referencing errors. No output written to miniperl |
> 
> If we could fix that, it would improve the value of the smoke reports.
> 

Fixed by adc48c40662e4c147b1f9b171574d14903affb00.

In the branch smoke-me/khw-uv, updated when I get around to it, I'm 
gradually trying to fix various issues like this.  That commit was made 
a week ago; ilmari++ figured out what the problem was after I brought it 
up on #irc.

Unfortunately, that doesn't fix these solaris smokes; they get a little 
further before breaking.  I was deferring pushing it until I had a 
complete solution, but since you brought it up, I pushed it.
And, since you did bring it up, I've just pushed another commit that 
fixes the next problem in line.

6c8e1158f1772eedb813932444244e833799a032, also diagnosed by ilmari++.

Now these solaris builds are blocked by Configure improperly having 
figured out the wrong type for the first argument to gethostbyaddr(). 
The uv branch has a 'set -x' in the salient shell code that does this.

Since I don't have access to a solaris box, this is slow going.  It 
takes several days generally for the results to come back, and I've made 
mistakes, and did the set -x in the wrong place.

In this branch, beyond solaris, I'm trying to work through the warnings 
that are generated by our smoke-windows-msvc142-amd64 github action 
(sample build report: https://github.com/Perl/perl5/runs/401268667).
There are gazillions of warnings there, many of which I think might bite 
us for values that take up 32 bits-ish.

If someone else also wants to work on these, that would be nice.
